### DISCUSSION TITLE
Call get_serializer TWICE in ModelMixins

### DISCUSSION THREAD

**Discussion by LeOndaz** [encode/django-rest-framework#9190] [Ideas & Suggestions]
Here's something that can improved, I was trying to have different serializers for input and for output, I overrode `create(...)` and it's fine. I found an improvement we could discuss that will make it super easy without interface change.

Here's the current implementation of model mixins
```
    def create(self, request, *args, **kwargs):
        serializer = self.get_serializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        self.perform_create(serializer)
        headers = self.get_success_headers(serializer.data)
        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
```
And the rest follow the same way, call the serializer, ..., return Response.

However, If we change it to
```
    def create(self, request, *args, **kwargs):
        serializer = self.get_serializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        self.perform_create(serializer)
        headers = self.get_success_headers(serializer.data)
        return Response(self.get_serializer(instance), status=status.HTTP_201_CREATED, headers=headers)
```

we would be able to do this:

```
def get_serializer(*args, **kwargs):
    data = kwargs.pop("data", None)

    if data is None:
        return output_serializer

    return input_serializer
```

What do you think?

**Comment by jeremystretch** [comment]
This seems like a bad idea. Not only would it likely be confusing, you're evading validation with the second call to `get_serializer()`. There's no guarantee that data which validates against the first serializer instance is also valid for the second.
